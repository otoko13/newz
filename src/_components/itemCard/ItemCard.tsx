import React from 'react';
import { INewsItem } from '../../NewzApp';
import './itemCard.scss';
import { Stack } from 'office-ui-fabric-react/lib/Stack';

export interface IItemCardProps {
    newsItem: INewsItem;
    isLatest: boolean;
    onItemSelected: () => void;
    index: number;
    visited: boolean;
}

const CARD_HEIGHT = 95;
const CARD_SPACING = 20;
export const CARD_BOX_HEIGHT = CARD_HEIGHT + CARD_SPACING;

const ItemCard = (props: IItemCardProps) => {
    return (
        <div 
            className={`ItemCard ${props.visited ? 'visited' : ''}`}
            onClick={props.onItemSelected} 
            style={
                { 
                    top: `${props.index * CARD_BOX_HEIGHT}px`,
                    marginBottom: `${CARD_SPACING}px`,
                    height: `${CARD_HEIGHT}px`
                }
            }>
            <Stack verticalFill verticalAlign='space-between'>
                <Stack.Item>
                    <div className='title'>{props.newsItem.title}</div>
                </Stack.Item>
                <Stack.Item>
                    <Stack horizontal horizontalAlign='space-between' className='details'>
                        <Stack.Item className='date'>
                            {props.newsItem.time}
                        </Stack.Item>
                        <Stack.Item className='author'>
                            {props.newsItem.by}
                        </Stack.Item>
                    </Stack>
                </Stack.Item>
            </Stack>
        </div>
    );
}

export default ItemCard;