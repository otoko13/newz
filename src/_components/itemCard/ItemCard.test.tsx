import { fireEvent, render } from '@testing-library/react';
import * as React from 'react';
import ItemCard, { IItemCardProps } from './ItemCard';

let props: IItemCardProps;

beforeEach(() => {
    props = {
        index: 3,
        isLatest: false,
        onItemSelected: jest.fn(),
        newsItem: {
            id: 1,
            descendants: 0,
            by: 'King Julian',
            kids: [2, 3, 4],
            time: 1581442689,
            title: 'A comment',
            text: 'I believe',
            type: 'story',
        },
        visited: false,
        selected: false,
    }
});

describe('snapshot', () => {
    it('should look sensible', () => {
        const { container } = render(<ItemCard {...props} />);
        expect(container.firstChild).toMatchSnapshot();
    });
});

describe('selecting item', () => {
    it('should call props.onItemSelected', () => {
        const { container } = render(<ItemCard {...props} />);
        expect(props.onItemSelected).not.toHaveBeenCalled();
        fireEvent.click(container.children[0]);
        expect(props.onItemSelected).toHaveBeenCalled();
    });
});