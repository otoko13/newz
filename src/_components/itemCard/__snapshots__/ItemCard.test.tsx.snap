// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot should look sensible 1`] = `
<div
  class="ItemCard   "
  style="top: 345px; margin-bottom: 20px; height: 95px;"
>
  <div
    class="ms-Stack css-36"
  >
    <div
      class="ms-StackItem css-37"
    >
      <div
        class="ms-Stack css-38"
      >
        <div
          class="ms-StackItem css-37"
        >
          <div
            class="title"
          >
            A comment
          </div>
        </div>
        <div
          class="ms-StackItem icons css-37"
        >
          <i
            aria-hidden="true"
            class="title title-icon root-33"
            data-icon-name="Articles"
          />
        </div>
      </div>
    </div>
    <div
      class="ms-StackItem css-37"
    >
      <div
        class="ms-Stack details css-39"
      >
        <div
          class="ms-StackItem date css-37"
        >
          17:38, Tuesday February 11th 2020
        </div>
        <div
          class="ms-StackItem author css-37"
        >
          King Julian
        </div>
      </div>
    </div>
  </div>
</div>
`;
